my-vault/internal/handlers/handlers.go

```diff
@@ -87,77 +82,75 @@ func HandlePostSecret(w http.ResponseWriter, r *http.Request) {
 		http.Error(w, err.Error(), http.StatusBadRequest)
 		return
 	}
-
+	// build new secretObject
 	temp := NewCreateSecret(secretRequestObject)
-	mu.Lock()
-	db[temp.Id] = temp
-	mu.Unlock()
-	
+	err = db.PutNewSecret(temp)
+
+	if err != nil {
+		
+		err := errors.New("unexpected internal error")
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
```

Comment:
Consider logging the error too so you see what happened
```suggestion
		err := errors.New("unexpected internal error")
		http.Error(w, err.Error(), http.StatusInternalServerError)
		fmt.Println(err)
		return
```